### ISSUE TITLE
Update Facility Locator template with components (v3 & V1)

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/va.gov-cms#17071]
## Description
The Facility Locator needs the following updates to components:

[Facility Locator](https://www.va.gov/find-locations)

### V1 components that need to be upgraded (Actionable)
1. Breadcrumbs
   - These are currently using the v1 component but they should be using [the v3 component](https://design.va.gov/components/breadcrumbs#examples---v3)
2. Modal
   - The modal appears if the user clicks "Use my location" but location sharing is not turned on in the browser
   - It is currently using the v1 component but they should be using [the v3 component](https://design.va.gov/components/modal#examples---v3)
3. Pagination
   - Currently using the v1 component but they should be using [the v3 component](https://design.va.gov/components/pagination#examples---v3)
4. Alert
   - v1 slim alerts are used for facility statuses. These should use [the v3 slim alerts](https://design.va.gov/components/alert#examples---slim-alert---v3)

### V1 components without V3 equivalents (Icebox)
1. Loading indicator: Currently using [v1](https://design.va.gov/components/loading-indicator) which is the most up to date component

### Elements that are not using components that should be (Backlog)
**These should be updated with the Facility Locator redesign that is already in progress**

1. Link
   - ~Directions links and~ other search tools link should be using [the v1 component](https://design.va.gov/components/link/) 
2. ~Button~
   - ~Buttons within the search filter should be using [the v3 component](https://design.va.gov/components/button/#examples---v3)~
3. Select
   - Form field within the search filters should use [the v3 component](https://design.va.gov/components/form/select#examples---v3)
4. ~Text input~
   - ~Form field within the search filters should use [the v3 component](https://design.va.gov/components/form/text-input#examples---v3)~
6. Telephone
   - These are using the proper telephone component but should be updated to use the [aria-desribedby variant](https://design.va.gov/components/telephone#aria-described-by)
   - Confirm exts are being parsed correctly within the component
7. Alert
   - The modal (mentioned above) looks like an alert but it is not using [an alert component](https://design.va.gov/components/alert#examples---standard---v3), should it? Can an alert component even be embedded inside a modal? 

## Acceptance Criteria
- [x] Breadcrumb component has been updated to v3
- [x] Pagination component has been updated to v3
- [x] Modal component has been updated to v3
- [x] Link component has been added
- [x] Button component has been added
- [x] Select component has been added
- [x] Text input component has been added
- [x] Telephone component has been update to aria-describedby variant
- [x] Exts in telephone numbers are being parsed correctly
- [x] Alert component has been added to modal if possible
- [x] Update and run existing test(s).
- [x] If additional tests are needed, stub a ticket to note this for future work.
- [ ] Requires accessibility review
- [ ] Provide notes on ngrok set up in ticket comment

**Comment by laflannery** [comment]
This is a [slack post about v3 breadcrumbs](https://dsva.slack.com/archives/C03R5SBELQM/p1707158519701019). It mentions documentation that should be followed when migrated to v3 so I think this might be helpful when this ticket does get pulled in

**Comment by maxx1128** [comment]
I estimate this at either a **5** or an **8.** It has many components to update, and the facility locator makes it a little more complicated.

**Comment by maxx1128** [comment]
**Update for end of Sprint 105:** I'm aiming to get the pull request for this ready for review by the end of the day today. Final reviews and closing will likely carry into next spring.

**Comment by maxx1128** [comment]
**End of Spring 106 update:** The tests are passing, and the AC is set, so review is the only thing left. I've been trying to get a tugboat environment ready for it, but it's been having build and asset issues, so this final step has been blocked for now.

**Comment by jilladams** [comment]
@maxx1128 are you able to provide a screenshot from your local? If you installed that Show/Hide VA Components plugin, and can screen cap it, that may suffice.

**Comment by laflannery** [comment]
My 2 cents - There were a decent amount of changes here and if we were able to get physical testing done prior to push that would be my preference because from doing this with PW there have been odd things I've run into during QA. Randi set up an ngrok for testing a couple times when tugboat was really struggling - would that be possible here?

**Comment by jilladams** [comment]
From planning: ngrok provides a URL others can use to point to local, for testing on mobile, etc. Eli knows how to get it working and can help provide pointers. (Also Randi & Daniel do.)

**Comment by laflannery** [comment]
Crossed out a couple things from the initial ticket:
- Directions links currently open in a new tab and component doesn't support that. The scope of this ticket does not change that behavior so for now we aren't doing that
- Text input does not support the "Clear input" behavior that is currently on the City/State field so we shouldn't change that right now either. We should handle this within FL locator redesign project

**Comment by jilladams** [comment]
Sprint 106 end: Laura has added some a11y / review feedback in PR. @maxx1128  will address that prior to merge; will close in next sprint.

**Comment by laflannery** [comment]
Noting from [Max's comment here](https://github.com/department-of-veterans-affairs/vets-website/pull/28433#issuecomment-2030214736) that the button could not be converted

**Comment by laflannery** [comment]
@jilladams I have also confirmed all these on prod but I know we have the outstanding #17807 so not sure if this can be closed or not

**Comment by jilladams** [comment]
Yep, that fix also deployed today and just ac / final checks are outstanding. So: closing this.
